### PR TITLE
Issue 4 fixes

### PR DESCRIPTION
**Pull request by vandadnp**

### CODE REVIEW

File: issue-4-functions-in-dart/issue-4-functions-in-dart.md [hunk 1]
```diff
@@ -68,62 +68,65 @@ minus the `mov eax 0xdeadbeef`, that's just the value to printed later, but the
...
 
-so until this point, before the `loc_9a63f` label, the 64-bit CPU register `eax` (the lower 32-bits of `rax`) is holding the value we are going to print to the console using the `print` function.
+so until this point, before the `loc_9a63f` label, the 32-bit CPU register `eax` (the lower 32-bits of `rax`) is holding the value we are going to print to the console using the `print` function. It's good to know, for me as well, that while operating in a 64-bit environment, a `mov` instruction onto a 32-bit register zeroes out the upper 32 bits of its 64-bit register, so that `mov eax, foo` will zero out the upper 32-bits of `rax`, `mov ebx, foo` will zero out the upper 32-bits of `rbx` and etc. I tried to find information about this in *Intel® 64 and IA-32 Architectures Software Developer’s Manual Volume 2 (2A, 2B, 2C & 2D): Instruction Set Reference, A-Z with no luck*. If you have a reference to this, I would love to have a tip please so we can add it to the bottom of this issue.
```
Comment (mraleph) [inline issue-4-functions-in-dart/issue-4-functions-in-dart.md]: It's in the section 3.4.1.1 in the Volume 1, which contains the following text (highlighted the relevant bullet point): 

> When in 64-bit mode, operand size determines the number of valid bits in the destination general-purpose register:
> • 64-bit operands generate a 64-bit result in the destination general-purpose register.
> • __32-bit operands generate a 32-bit result, zero-extended to a 64-bit result in the destination general-purpose register.__
> • 8-bit and 16-bit operands generate an 8-bit or 16-bit result. The upper 56 bits or 48 bits (respectively) of the destination general-purpose register are not modified by the operation. If the result of an 8-bit or 16-bit operation is intended for 64-bit address calculation, explicitly sign-extend the register to the full 64-bits.

File: issue-4-functions-in-dart/issue-4-functions-in-dart.md [hunk 2]
```diff
@@ -200,6 +203,8 @@ void main(List<String> args) {
 }
 ```
 
+Note: we are not actually invoking the foo function, rather printing its descriptor!
```
Comment (mraleph) [inline issue-4-functions-in-dart/issue-4-functions-in-dart.md]: To be more precise we print `foo`'s tear-off - which is just a closure forwarding invocations to `foo`. 

Compare:

```dart
void foo() { }

void main() {
  var f = foo; // foo's closurisation also known as tear-off.
  print(f); // closures are just normal objects, so you can print them

  var f_toString = f.toString;  // you can also tear off methods on objects
                                // you get a closure which invokes method on a fixed object
  print(f_toString());  // the same as calling f.toString() here.
}
```

**Review by mraleph — APPROVED** [review]

**Comment by vandadnp** [comment]
@mraleph I made some more corrections, would you be able to please have a look at them when and if you have the time? Much appreciated

**Comment by mraleph** [comment]
looks good to me.

**Comment by vandadnp** [comment]
Thank you so very much, for your expertise and your time, @mraleph, I really appreciate it. It's so refreshing to see people like you at such high levels trying to help out the community.

**Pull request merged**